packages/x-data-grid/src/components/columnHeaders/GridColumnHeaderItem.tsx

```diff
@@ -287,6 +279,15 @@ function GridColumnHeaderItem(props: GridColumnHeaderItemProps) {
 
   const label = colDef.headerName ?? colDef.field;
 
+  const style = React.useMemo(() => {
+    let styleProp = props.style;
+    const pinnedSide = rtlFlipSide(pinnedPosition, isRtl);
+    if (pinnedSide && pinnedOffset !== undefined) {
+      styleProp = { ...styleProp, [pinnedSide]: pinnedOffset };
+    }
+    return styleProp;
```

Comment:
I see this block of code repeated a few times, maybe it wants to be a function?

I'm also curious why those styles are now a bit everywhere, where were they before?

Reply:
The styles were previously defined on parent level, and passed as props as unstable objects, which broke the memoizations.

Reply:
I mean we could simplify it to:

```
const styleProp = attachPinnedStyle(props.style, pinnedPosition, pinnedOffset)
```

Because I see this block at least 3-4 times I think.